cmd/kubelet/app/server.go

```diff
@@ -160,6 +161,15 @@ HTTP server: The kubelet can also listen for HTTP and respond to a simple API
 				os.Exit(1)
 			}
 
+			// logging config should be validate and apply early
+			if errs := logs.ValidateLoggingConfiguration(&kubeletConfig.Logging, field.NewPath("logging")); len(errs) > 0 {
+				klog.ErrorS(err, "Failed to validate kubelet logging config")
+				os.Exit(1)
+			}
+
+			logOption := &logs.Options{Config: kubeletConfig.Logging}
+			logOption.Apply()
```

Comment:
doesn't moving this here mean that if the logging options are provided in the kubelet config file (vs in CLI flags), they don't take affect?

Reply:
Log format options should be validate and apply ealier, if not some log lines will be not follow log format set in options.

Reply:
did you test providing the logging configuration in the kubelet config file and ensuring it took effect?

Reply:
I think this log line will be missed.

https://github.com/kubernetes/kubernetes/blob/master/cmd/kubelet/app/server.go#L120-L122

Would it be possible to return an error of invalid logging configration type wen NewKubeletConfiguration is called and then if the logging configuration is invalid; then we exit with a non-json log line. Otherwise, the json logging validation is applied.

Reply:
@manugupt1 Same dicussion in here https://github.com/kubernetes/kubernetes/pull/100409#discussion_r657152114